databases/backends/common/records.py

```diff
@@ -0,0 +1,142 @@
+import json
+import typing
+from datetime import date, datetime
+
+from sqlalchemy.engine.interfaces import Dialect
+from sqlalchemy.engine.row import Row as SQLRow
+from sqlalchemy.sql.compiler import _CompileLabel
+from sqlalchemy.sql.schema import Column
+from sqlalchemy.types import TypeEngine
+
+from databases.interfaces import Record as RecordInterface
+
+DIALECT_EXCLUDE = {"postgresql"}
+
+
+class Record(RecordInterface):
+    __slots__ = (
+        "_row",
+        "_result_columns",
+        "_dialect",
+        "_column_map",
+        "_column_map_int",
+        "_column_map_full",
+    )
+
+    def __init__(
+        self,
+        row: typing.Any,
+        result_columns: tuple,
+        dialect: Dialect,
+        column_maps: typing.Tuple[
+            typing.Mapping[typing.Any, typing.Tuple[int, TypeEngine]],
+            typing.Mapping[int, typing.Tuple[int, TypeEngine]],
+            typing.Mapping[str, typing.Tuple[int, TypeEngine]],
+        ],
+    ) -> None:
+        self._row = row
+        self._result_columns = result_columns
+        self._dialect = dialect
+        self._column_map, self._column_map_int, self._column_map_full = column_maps
+
+    @property
+    def _mapping(self) -> typing.Mapping:
+        return self._row
+
+    def keys(self) -> typing.KeysView:
+        return self._mapping.keys()
+
+    def values(self) -> typing.ValuesView:
+        return self._mapping.values()
+
+    def __getitem__(self, key: typing.Any) -> typing.Any:
+        if len(self._column_map) == 0:
+            return self._row[key]
+        elif isinstance(key, Column):
+            idx, datatype = self._column_map_full[str(key)]
+        elif isinstance(key, int):
+            idx, datatype = self._column_map_int[key]
+        else:
+            idx, datatype = self._column_map[key]
+
+        raw = self._row[idx]
+        processor = datatype._cached_result_processor(self._dialect, None)
+
+        if self._dialect.name not in DIALECT_EXCLUDE:
+            if isinstance(raw, dict):
+                raw = json.dumps(raw)
+
+        if processor is not None and (not isinstance(raw, (datetime, date))):
+            return processor(raw)
+        return raw
+
+    def __iter__(self) -> typing.Iterator:
+        return iter(self._row.keys())
```

Comment:
Because of this you can't unpack result
Example
```
class LastUsername(Base):
	__tablename__ = 'last_username'
	user_id = Column(BIGINT, primary_key=True)
	username = Column(TEXT, nullable=False)

r = await db.fetch_one(select(LastUsername))
user_id, username = r
assert user_id == 'user_id' #but it need to be at least int
```